exec/cnex/cell.c

```diff
@@ -596,7 +596,7 @@ int32_t cell_compareCell(const Cell * s, const Cell * d)
 
     switch (s->type) {
         case cAddress:      return s->address != d->address;
-        case cArray:        return s->array->size != d->array->size;
+        case cArray:        return array_compareArray(s->array, d->array);
```

Comment:
Comparing this to the dictionary case, one of these looks wrong. This case returns the result of `array_compareArray`, while the dictionary case returns the inverse of `dictionary_compareDictionary`. If these aren't wrong, then the functions should be fixed to return consistent values.

Reply:
Fixed array compare logic so that it matches the dictionary matching logic.